gap/attr.gi

```diff
@@ -1687,6 +1687,138 @@ function(D)
   return fail;
 end);
 
+InstallMethod(DigraphCore, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local N, lo, topdown, bottomup, hom, lo_var, image,
+  comps, comp, cores, G, vert_labels, D, in_core, n, m, L;
+  if not IsImmutableDigraph(digraph) then
+    digraph := DigraphImmutableCopy(digraph);
+  fi;
+  N := DigraphNrVertices(digraph);
+  if IsEmptyDigraph(digraph) then
+    if N >= 1 then
+      return [1];
+    else
+      return [];
+    fi;
+  fi;
+  digraph := ReducedDigraph(digraph);  # isolated verts are not in core
+  if DigraphHasLoops(digraph) then
+    return DigraphLoops(digraph){[1]};
+  elif not IsConnectedDigraph(digraph) then
+    comps := DigraphConnectedComponents(digraph).comps;
+    DigraphVertexLabels(digraph);
+    cores := [];
+    for comp in comps do
+      D := InducedSubdigraph(digraph, comp);
+      D := InducedSubdigraph(D,
+           Positions(BlistList(DigraphVertexLabels(D), DigraphCore(D)), true));
+      Add(cores, D);
+    od;
+    L := Length(cores);
+    in_core := List([1 .. L], x -> true);
+    n := 1;
+    while n <= L do
+      if n <> L then
+        m := n + 1;
+      else
+        m := 1;
+      fi;
+      while m <> n and in_core[n] do
+        if in_core[m] and DigraphHomomorphism(cores[m], cores[n]) <> fail then
+          in_core[m] := false;
+        fi;
+        if m < L then
+          m := m + 1;
+        else
+          m := 1;
+        fi;
+      od;
+      n := n + 1;
+    od;
+    cores := ListBlist(cores, in_core);
+    vert_labels := [];
+    for G in cores do
+      vert_labels := Concatenation(vert_labels, DigraphVertexLabels(G));
+    od;
+    return vert_labels;
+  elif IsCompleteDigraph(digraph) then
+    return [1 .. N];
+  elif IsSymmetricDigraph(digraph) and IsBipartiteDigraph(digraph) then
+    return DigraphEdges(digraph)[1];
```

Comment:
This is a potentially very slow way to get a single edge of a digraph - we always want to avoid constructing the list/set of edges if possible, it can be slow to produce and take up a lot of memory.

It would take more code, but you could do something like:
```
i := First(DigraphVertices(digraph), i -> OutDegreeOfVertex(digraph, i) > 0);
return [i, OutNeighboursOfVertex(i, 1)];
```
instead.